core/src/test/java/org/owasp/dependencycheck/analyzer/OssIndexAnalyzerTest.java

```diff
@@ -252,18 +251,20 @@ void should_analyzeDependency_fail_when_socket_error_from_sonatype() throws Exce
     }
 
     @Test
-    void should_prepareAnalyzer_fail_when_credentials_not_set() throws Exception {
+    void should_prepareAnalyzer_disable_when_credentials_not_set() throws Exception {
+        // Given
         OssIndexAnalyzer analyzer = new OssIndexAnalyzer();
         Settings settings = getSettings();
         Engine engine = new Engine(settings);
         analyzer.initialize(settings);
-        try {
-            analyzer.prepareAnalyzer(engine);
-            assertThrows(InitializationException.class, () -> analyzer.prepareAnalyzer(engine));
-        } catch (InitializationException e) {
-          analyzer.close();
-          engine.close();
-        }
+
+        // When
+        analyzer.prepareAnalyzer(engine);
+
+        // Then
+        assertFalse(analyzer.isEnabled());
+        analyzer.close();
+        engine.close();
```

Comment:
One way would be to just flip things around like this:

```suggestion
        boolean enabled = analyzer.isEnabled();
        analyzer.close();
        engine.close();
        assertFalse(enabled);
```

A better approach might be to utilize a try with resources:

```java
        OssIndexAnalyzer analyzer = new OssIndexAnalyzer();
        Settings settings = getSettings();
        try (Engine engine = new Engine(settings)) {
                analyzer.initialize(settings);   
                // When
                analyzer.prepareAnalyzer(engine);
                // Then
                assertFalse(analyzer.isEnabled());
        } finally {
                analyzer.close();
        }
```